src/kotlinSessions/SessionTwo.kt

```diff
@@ -55,10 +59,99 @@ fun main() {
     for (number in range step 1) {
         println("Before continue")
         if (number > 3) {
+            println("Inside the condition !")
             // skips the lines after it and continue looping
             continue
         }
         println("After continue")
     }
+
     //-----------------------------------------------------------------------
+
+
+    //--------------------------------- Exercises --------------------------------------------
+
+    // The factorial function (symbol: !) refers to multiplying all whole numbers from our chosen number down to 1.
+    // Rule = n! = n × (n−1) x (n-2) x (n-3) x .........
+    // see more at : https://www.mathsisfun.com/numbers/factorial.html
+    // request user input for a number value
+    val number: String? = readLine()
+    // check if the number is available
+    if (number != null) {
+        //convert the input to a number
+        val input: Int = number.toInt()
+        // create an interval for the factorial
+        val range = 1..input
+        // create a memory to hold the result
+        var result: Int = 1
+        // find the product of this range
+        for (n in range) {
+            result = n * result
+        }
+        println(result)
+    } else {
+        println("Invalid Input !")
+    }
+
+    // equivalent code
+
+    // !! is the 'not-null assertion operator'
+    // it checks if the value is null, if so then it converts it to a non-null value or empty string "".
+    // it asserts the value to be non-null.
+    // and its equivalent to
+    /*
+     *
+     *  var number: String? = readLine()
+     *  if (number != null) {
+     *     number = number
+     *  } else {
+     *     number = ""
+     *  }
+     *
+     */
+    val input: Int = readLine()!!.toInt()
+    // create an interval for the factorial
+    // create a memory to hold the result
+    var result: Int = 1
+    // find the product of this range
+    for (n in 1..input) {
+        result = n * result
+    }
+    println(result)
+
+    /*
+    * Debugging exercise (DIY) :
+    *
+    * In the previous examples, try to input non-number values such as : your
+    * name for example, observe and record your output and try to find a solution.
+    *
+    * Send the solution back to us !
+    *
+    */
+
+    // multiplication table examples
+
+    // In this example, we will guide you step-by-step to build a CLI program to print the
+    // multiplication table of a given number.
+
+    // 1) Request user input.
+    val userInput: String = readLine()!!
+    // 2) Create a variable to hold some data
```

Comment:
what hold some data means

Reply:
A `temporary` or `temp` value.

I will change the vague docs.